images/fluent-watcher/main.tf

```diff
@@ -0,0 +1,43 @@
+locals {
+  extra_packages = [
+    "fluent-bit", "fluent-bit-compat", "fluent-watcher-compat"
+  ]
+}
+
+variable "target_repository" {
+  description = "The docker repo into which the image and attestations should be published."
+}
+
+module "versions" {
+  package = "fluent-watcher"
+  source  = "../../tflib/versions"
+}
+
+module "config" {
+  for_each       = module.versions.versions
+  source         = "../fluent-operator/config"
```

Comment:
also here, we should create our own config for fluent-watcher according to the Dockerfile for the project:

- https://github.com/fluent/fluent-operator/blob/v2.9.0/cmd/fluent-watcher/fluentbit/Dockerfile

Reply:
So the whole idea of the watcher is that it can detect changes to the fluent-bit config.
So, maybe a test like : https://github.com/chainguard-images/images/blob/main/images/fluent-bit/tests/main.tf

That deploys fluent-bit but with an overriden entrypoint which won't run the watcher. So edit the values:
```
image:
  repository: 
  tag: 
  digest: 
command:
  - /fluent-bit/bin/fluent-bit-watcher 
args:
  - -c=/fluent-bit/etc/conf/fluent-bit.conf  
  - -watch-path=/fluent-bit/etc/conf
  ```

That creates a ConfigMap called <chart name>-fluent-bit with a fluent-bit.conf section. Replace that section with something a bit different and monitor the log file for the message when it changes:
```
level=info time=2024-06-21T10:23:03Z msg="Config file changed, reloading..."                                                                                                                                                                                                                    
```

I'd recommend not using the "default" config because it tries to connect to elastic and fails and there is a lot of noise in the log from that and it's really slow. eg in values.yaml:
```
config:
  outputs: []
```

You can't change the configmap by changing the values and rerunning helm because that restarts the pod! You have to change the CM direct. (For example, remove the whole "FILTERS" section.